fpdf/image_parsing.py

```diff
@@ -48,20 +51,291 @@ def _decode_base64_image(base64Image):
+    0xCC,
+    0x2C,
+    0xAC,
+    0x6C,
+    0xEC,
+    0x1C,
+    0x9C,
+    0x5C,
+    0xDC,
+    0x3C,
+    0xBC,
+    0x7C,
+    0xFC,
+    0x02,
+    0x82,
+    0x42,
+    0xC2,
+    0x22,
+    0xA2,
+    0x62,
+    0xE2,
+    0x12,
+    0x92,
+    0x52,
+    0xD2,
+    0x32,
+    0xB2,
+    0x72,
+    0xF2,
+    0x0A,
+    0x8A,
+    0x4A,
+    0xCA,
+    0x2A,
+    0xAA,
+    0x6A,
+    0xEA,
+    0x1A,
+    0x9A,
+    0x5A,
+    0xDA,
+    0x3A,
+    0xBA,
+    0x7A,
+    0xFA,
+    0x06,
+    0x86,
+    0x46,
+    0xC6,
+    0x26,
+    0xA6,
+    0x66,
+    0xE6,
+    0x16,
+    0x96,
+    0x56,
+    0xD6,
+    0x36,
+    0xB6,
+    0x76,
+    0xF6,
+    0x0E,
+    0x8E,
+    0x4E,
+    0xCE,
+    0x2E,
+    0xAE,
+    0x6E,
+    0xEE,
+    0x1E,
+    0x9E,
+    0x5E,
+    0xDE,
+    0x3E,
+    0xBE,
+    0x7E,
+    0xFE,
+    0x01,
+    0x81,
+    0x41,
+    0xC1,
+    0x21,
+    0xA1,
+    0x61,
+    0xE1,
+    0x11,
+    0x91,
+    0x51,
+    0xD1,
+    0x31,
+    0xB1,
+    0x71,
+    0xF1,
+    0x09,
+    0x89,
+    0x49,
+    0xC9,
+    0x29,
+    0xA9,
+    0x69,
+    0xE9,
+    0x19,
+    0x99,
+    0x59,
+    0xD9,
+    0x39,
+    0xB9,
+    0x79,
+    0xF9,
+    0x05,
+    0x85,
+    0x45,
+    0xC5,
+    0x25,
+    0xA5,
+    0x65,
+    0xE5,
+    0x15,
+    0x95,
+    0x55,
+    0xD5,
+    0x35,
+    0xB5,
+    0x75,
+    0xF5,
+    0x0D,
+    0x8D,
+    0x4D,
+    0xCD,
+    0x2D,
+    0xAD,
+    0x6D,
+    0xED,
+    0x1D,
+    0x9D,
+    0x5D,
+    0xDD,
+    0x3D,
+    0xBD,
+    0x7D,
+    0xFD,
+    0x03,
+    0x83,
+    0x43,
+    0xC3,
+    0x23,
+    0xA3,
+    0x63,
+    0xE3,
+    0x13,
+    0x93,
+    0x53,
+    0xD3,
+    0x33,
+    0xB3,
+    0x73,
+    0xF3,
+    0x0B,
+    0x8B,
+    0x4B,
+    0xCB,
+    0x2B,
+    0xAB,
+    0x6B,
+    0xEB,
+    0x1B,
+    0x9B,
+    0x5B,
+    0xDB,
+    0x3B,
+    0xBB,
+    0x7B,
+    0xFB,
+    0x07,
+    0x87,
+    0x47,
+    0xC7,
+    0x27,
+    0xA7,
+    0x67,
+    0xE7,
+    0x17,
+    0x97,
+    0x57,
+    0xD7,
+    0x37,
+    0xB7,
+    0x77,
+    0xF7,
+    0x0F,
+    0x8F,
+    0x4F,
+    0xCF,
+    0x2F,
+    0xAF,
+    0x6F,
+    0xEF,
+    0x1F,
+    0x9F,
+    0x5F,
+    0xDF,
+    0x3F,
+    0xBF,
+    0x7F,
+    0xFF,
+]
+
+
+def get_img_info(filename, imgbytesio=None, image_filter="AUTO", dims=None):
     """
     Args:
-        img: `BytesIO` or `PIL.Image.Image` instance
+        filename: in a format that can be passed to load_image_content
+        imgbytesio: optional `BytesIO` or `PIL.Image.Image` instance
         image_filter (str): one of the SUPPORTED_IMAGE_FILTERS
     """
     if Image is None:
         raise EnvironmentError("Pillow not available - fpdf2 cannot insert images")
 
-    if not isinstance(img, Image.Image):
-        img = Image.open(img)
+    imgreadable = None
+    img = None
+    if not imgbytesio or isinstance(imgbytesio, Path):
+        imgreadable = load_image(filename)
+        img = Image.open(imgreadable)
```

Comment:
So in 2 cases in this `if` the image will be opened twice?
That's a shame performance-wise

Reply:
I'm not exactly sure what you mean? in my understanding the file will be opened just once, and `Image.open` is lazy loading the image so it doesn't decode the image if it doesn't need to (and in the code I added I only look at metadata so it's very fast)

Reply:
Yes I was a bit unclear. You are totally right, `Image.open` is indeed lazy ([source](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.open)):
> This is a lazy operation; this function identifies the file, but the file remains open and the actual image data is not read from the file until you try to process the data (or call the [load()](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.load) method).

The image will be read at some point as your code then accesses the following properties of `img`:
* `img.mode`
* `img.format`
* `img.info["compression"]`
* `img.tag_v2`

What I missed is that `Image.open` is not called elsewhere anymore, right?
So I do not see any potential performance problem 😊

Reply:
great! I don't think it's used anywhere no, my understanding is that the pixel map will not be read so there shouldn't be any performance problem